client/sections-middleware.js

```diff
@@ -90,6 +98,10 @@ function createPageDefinition( path, sectionDefinition ) {
 					dispatch( { type: 'SECTION_SET', isLoading: false } );
 					LoadingError.show( context, sectionDefinition.name );
 				}
+			} )
+			.then( () => {
```

Comment:
This last `then` handler is also executed in case when the chunk loading fails. The `catch` handler will return a resolved promise that can be further chained. Promises are awesome for chaining like this.

Reply:
Thank you! I appreciate the explanation